### PR TITLE
42 interface for post processor output

### PR DESCRIPTION
**Pull request by miklasr**
# Details

- Add interface for the post processor
- Updates structure for namespacing our system blocks
- Add copyright GPLv3 license statement to the top of all source files

# Checklist

## Code
- [ ] ~I have added some tests~
- [ ] ~I have run the tests locally to ensure they all pass~

## Documentation
- [x] I have updated documentation appropriately (GitHub Pages)
- [x] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)
- [x] I have documented header files for use with Doxygen

Works on  #42

### CODE REVIEW

File: src/post_processor.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  /**
+   * @brief Set the confidence threshold
+   *
+   * @param confidence_threshold The updated confidence threshold in the range
+   * [0, 1]
+   * @return true If updating threshold was successful
+   * @return false If the new threshold value is invalid
+   */
+  bool set_confidence_threshold(float confidence_threshold);
```
Comment (Ashwin-MJ) [inline src/post_processor.h:121]: Would we have a private variable `confidence_threshold` for this?

Comment (miklasr) [reply src/post_processor.h:121]: Thanks! Added that.

**Review by Ashwin-MJ — APPROVED** [review]
Looks good to me! Using namespaces should make things a lot tidier

File: src/post_processor.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SRC_POST_PROCESSOR_H_
+#define SRC_POST_PROCESSOR_H_
+
+#include "inference_core.h"
+
+namespace PostProcessing {
```
Comment (C-Begley) [inline src/post_processor.h:28]: "Namespace names are all lower-case, with words separated by underscores." - from Google Style Guide

**Comment by C-Begley** [comment]
cpplint uses Google Style Guide right? Because it states "Namespace names are all lower-case, with words separated by underscores.", so may need to change your namespacing

**Comment by miklasr** [comment]
> cpplint uses Google Style Guide right? Because it states "Namespace names are all lower-case, with words separated by underscores.", so may need to change your namespacing

cpplint doesn't complain about that. I think that's something that's up to us.

**Comment by miklasr** [comment]
> cpplint uses Google Style Guide right? Because it states "Namespace names are all lower-case, with words separated by underscores.", so may need to change your namespacing

I followed the style TFL had used, and that's a Google product ...

**Comment by C-Begley** [comment]
> I followed the style TFL had used, and that's a Google product ...

Fair enough, I was just following this : https://google.github.io/styleguide/cppguide.html#General_Naming_Rules

We should probably agree as a group which we will use, as all our namespacing should be the same

**Comment by miklasr** [comment]
> > I followed the style TFL had used, and that's a Google product ...
> 
> Fair enough, I was just following this : https://google.github.io/styleguide/cppguide.html#General_Naming_Rules
> 
> We should probably agree as a group which we will use, as all our namespacing should be the same

I would suggest CamelCase and I think Ashwin is of the same opinion. Would_that_be_alright?

**Comment by C-Begley** [comment]
> > > I followed the style TFL had used, and that's a Google product ...
> > 
> > 
> > Fair enough, I was just following this : https://google.github.io/styleguide/cppguide.html#General_Naming_Rules
> > We should probably agree as a group which we will use, as all our namespacing should be the same
> 
> I would suggest CamelCase and I think Ashwin is of the same opinion. Would_that_be_alright?

I personally prefer lower case, as the even the C++ standard library namespace is lower case. However, if majority rules, CamelCase is fine with me.

**Review by C-Begley — APPROVED** [review]
Happy to go with this and use it as a template for future header file standards

**Review by Andrew-Ritchie — APPROVED** [review]
Looks good, like the style used!

**Pull request merged**